roxctl/common/flags/env_flag_test.go

```diff
@@ -30,6 +31,145 @@ func TestFlagOrSettingValue(t *testing.T) {
 	assert.Equal(t, "some-test-value", Password())
 }
 
+// TODO: How do I make this work from running "test ."?
+func TestFlagOrConfigurationValueFlags(t *testing.T) {
+
+	var (
+		testFile1 = "./testdata/test_instance1.yaml"
+	)
+	// 1. Default, unchanged flag value and setting not set should lead to the default value being returned.
+	cmd := &cobra.Command{
+		PersistentPreRunE: LoadConfig,
+	}
+
+	AddAPITokenFile(cmd)
+	AddConnectionFlags(cmd)
+	AddConfigurationFile(cmd)
+
+	assert.False(t, ConfigurationFileChanged())
+	assert.False(t, *caCertFileSet)
+	assert.Empty(t, *apiTokenFileChanged)
+	assert.False(t, *endpointChanged)
+
+	err := cmd.PersistentFlags().Set("config-file", testFile1)
+	assert.NoError(t, err)
+	assert.Equal(t, testFile1, ConfigurationFileName())
+
+	err = cmd.PersistentPreRunE(cmd, []string{})
+	assert.True(t, ConfigurationFileChanged())
+
+	// Execute the command to trigger PersistentPreRunE
```

Comment:
What is the added value of this comment ? PersistentPreRunE is called 2 lines before.